pkg/collector/corechecks/agentprofiling/agentprofiling.go

```diff
@@ -0,0 +1,238 @@
+		}
+
+		// RSS (Resident Set Size) represents the total memory allocated to the process
+		// This includes all memory: Go heap, native libraries, and other allocations
+		processMemoryMB = float64(memInfo.RSS) / MB
+	}
+
+	// Get Agent CPU usage
+	var currentCPU float64
+	if m.instance.CPUThreshold > 0 {
+		// Get the current process (agent)
+		p, err := process.NewProcess(int32(os.Getpid()))
+		if err != nil {
+			return fmt.Errorf("Failed to get agent process: %s", err)
+		}
+
+		// Get the total CPU time used by the process
+		cpuTimes, err := p.Times()
+		if err != nil {
+			return fmt.Errorf("Failed to get agent CPU times: %s", err)
+		}
+
+		// Calculate CPU percentage since last check
+		currentCPU = m.calculateCPUPercentage(cpuTimes)
+	}
+
+	// Exit early if usage is below thresholds
+	if processMemoryMB < float64(m.memoryThreshold)/MB && currentCPU < float64(m.instance.CPUThreshold) {
+		log.Debugf("Memory and CPU usage are below thresholds (Memory: %.2f MB < %.2f MB, CPU: %.2f%% < %d%%), skipping Agent profiling check.", processMemoryMB, float64(m.memoryThreshold)/MB, currentCPU, m.instance.CPUThreshold)
+		return nil
+	}
+
+	// If either memory or CPU exceeds threshold, generate flare
+	log.Infof("Threshold exceeded - Memory: %.2f MB, CPU: %.2f%%. Generating flare.", processMemoryMB, currentCPU)
+	if err := m.generateFlare(); err != nil {
+		return fmt.Errorf("Failed to generate flare: %w", err)
+	}
+
+	return nil
+}
+
+// generateFlare generates a flare and sends it to Zendesk if ticketID is specified, otherwise generates it locally
+func (m *Check) generateFlare() error {
+	// Skip flare generation if flareComponent is not available
+	if m.flareComponent == nil {
+		log.Info("Skipping flare generation: flare component not available")
+		m.flareGenerated = true
+		return nil
+	}
+
+	// Prepare flare arguments
+	flareArgs := types.FlareArgs{
+		ProfileDuration:      m.agentConfig.GetDuration("flare.rc_profiling.profile_duration"),
+		ProfileBlockingRate:  m.agentConfig.GetInt("flare.rc_profiling.blocking_rate"),
+		ProfileMutexFraction: m.agentConfig.GetInt("flare.rc_profiling.mutex_fraction"),
+	}
+
+	// Create an instance of the flare struct
+	flarePath, err := m.flareComponent.CreateWithArgs(flareArgs, 0, nil, []byte{})
+	if err != nil {
+		return fmt.Errorf("Failed to create flare: %w", err)
+	}
+
+	if m.instance.TicketID != "" && m.instance.UserEmail != "" {
+		// Send the flare
+		caseID := m.instance.TicketID
+		userHandle := m.instance.UserEmail
+		response, err := m.flareComponent.Send(flarePath, caseID, userHandle, helpers.NewLocalFlareSource())
+		if err != nil {
+			// Add debugging logs to capture the response from Datadog backend
+			log.Errorf("Datadog backend response: %s", response)
+			return fmt.Errorf("Failed to send flare: %w", err)
+		}
+		log.Infof("Flare sent with case ID %s", m.instance.TicketID)
+	} else {
+		log.Infof("Flare generated locally at %s", flarePath)
```

Comment:
%q to ensure that the string is quoted and does not break the log output.

```suggestion
		log.Infof("Flare generated locally at %q", flarePath)
```